src/main/java/org/owasp/esapi/waf/ESAPIWebApplicationFirewallFilter.java

```diff
@@ -60,448 +58,436 @@
-        }
-
-    }
-
-    public void destroy() {
-        /*
-         * Any cleanup necessary?
-         */
-    }
-
-    private void sendRedirect(InterceptingHTTPServletResponse response, HttpServletResponse httpResponse)
-            throws IOException {
-        /*
-         * [chrisisbeef] - commented out as this is not currently used. Minor
-         * performance tweak. String finalJavaScript =
-         * AppGuardianConfiguration.JAVASCRIPT_REDIRECT; finalJavaScript =
-         * finalJavaScript.replaceAll(AppGuardianConfiguration.
-         * JAVASCRIPT_TARGET_TOKEN, appGuardConfig.getDefaultErrorPage());
-         */
-
-        if (response != null) {
-            response.reset();
-            response.resetBuffer();
-            /*
-             * response.setStatus(appGuardConfig.getDefaultResponseCode());
-             * response.getOutputStream().write(finalJavaScript.getBytes());
-             */
-            response.sendRedirect(appGuardConfig.getDefaultErrorPage());
-
-        } else {
-            if (!httpResponse.isCommitted()) {
-                httpResponse.sendRedirect(appGuardConfig.getDefaultErrorPage());
-            } else {
-                /*
-                 * Can't send redirect because response is already committed.
-                 * I'm not sure how this could happen, but I didn't want to
-                 * cause IOExceptions in case if it ever does.
-                 */
-            }
-
-        }
-    }
+	private AppGuardianConfiguration appGuardConfig;
+
+	private static final String CONFIGURATION_FILE_PARAM = "configuration";
+	private static final String LOGGING_FILE_PARAM = "log_settings";
+	private static final String POLLING_TIME_PARAM = "polling_time";
+
+	private static final int DEFAULT_POLLING_TIME = 30000;
+
+	private String configurationFilename = null;
+
+	private long pollingTime;
+
+	private long lastConfigReadTime;
+
+	// private static final String FAUX_SESSION_COOKIE = "FAUXSC";
+	// private static final String SESSION_COOKIE_CANARY =
+	// "org.owasp.esapi.waf.canary";
+
+	private FilterConfig fc;
+
+	private final Logger logger = ESAPI.getLogger(ESAPIWebApplicationFirewallFilter.class);
+
+	/**
+	 * This function is used in testing to dynamically alter the configuration.
+	 * 
+	 * @param policyFilePath
+	 *            The path to the policy file
+	 * @param webRootDir
+	 *            The root directory of the web application.
+	 * @throws FileNotFoundException
+	 *             if the policy file cannot be located
+	 */
+	public void setConfiguration(String policyFilePath, String webRootDir) throws FileNotFoundException {
+
+		FileInputStream inputStream = null;
+
+		try {
+			inputStream = new FileInputStream(new File(policyFilePath));
+			appGuardConfig = ConfigurationParser.readConfigurationFile(inputStream, webRootDir);
+			lastConfigReadTime = System.currentTimeMillis();
+			configurationFilename = policyFilePath;
+		} catch (ConfigurationException e) {
```

Comment:
If we're touching this line, we might as well fix the typo. Should be 'threw' rather than 'through'. (Probably my fault. I do that all the time and it sounds like one of my comments.)

Reply:
Easy enough.  This has been changed.